File: practical05/fibonacci.c
#include <stdio.h>

/* Declaring functions */
void updateValues(int *previous, int *current);

int main(void) {
/* Declaring variables local to main */
    int i, n;
    int series[200];
    int previous, current;    

/* User inputs integer to calculate series up to */
    printf("Input a non-zero positive integer:\n");
    scanf("%d", &n);
    
/* Initial value for series and integers, set up pointers */
    series[0] = 0;
    series[1] = 1;
    previous = 0;
    current = 1;

/* Loop to add values to series up to n */
    for (i=2; i < n; i++) {
        updateValues(&previous, &current);
        series[i] = current;
    }

/* Printing series values */
    printf("The Fibonacci sequence up to %d values is:\n", n);
    for (i = 0; i < n; i++) {
        printf("%d\n", series[i]);
    }

    return 0;
}

/* Function to update values of previous and current */
void updateValues(int *pre,int *cur) {
    int temp;

    temp = *pre;
    *pre = *cur;
    *cur = *cur + temp;

    return;
}

File: practical05/fibonacci.f90
program fibonacci
! Declaring variables
    integer (kind = 4) :: n, i, previous, current
    integer (kind = 4) :: series(50)

! Ask user for number of digits to calculate to
    write(6,*) 'Choose a non-zero positive integer:'
    read(5,*), n

! Set inital values
    previous = 0
    current = 1
    series(1) = 0
    series(2) = 1

! Loop to update series up to n digits
    i = 3
    do while (i < n+1)
        call updateValues(previous, current)
        series(i) = current
        i = i+1
    end do

! Print series
    write(*,*) 'The series up to ', n, ' digits is:'
    i = 1
    do while (i < n+1)
        write(*, *) series(i)
        i = i+1
    end do

end program fibonacci

! Subroutine to update current and previous values
subroutine updateValues(pre,cur)
    integer (kind = 4), intent(inout) :: pre, cur
    integer (kind = 4) :: temp

    temp = pre
    pre = cur
    cur = cur + temp
end subroutine updateValues
